### PR TITLE
Item durability loss now depends on the item used.

### LINKED ISSUE #4119: Dig block with an armor in hand damages the armor.
Client version:   x.x.x
Server OS:        Windows/Linux/macOS/BSD/...
Commit id:        (Found at the top of the server's console output)

### Expected behavior
Hold an armor in hand, dig an dirt block, the armor keeps the durability.

### Actual behavior
The armor lost one durability.

### Steps to reproduce the behavior
As described.

I cannot think of any case that the armor should be damaged in cPlayer::UseEquippedItem() function.

### PR DESCRIPTION
**Pull request by bearbin**
Armour durability also no longer changes when it is used to break blocks or attack mobs.

Fixes #4119

### CODE REVIEW

File: src/Items/ItemHandler.h [hunk 1]
```diff
@@ -21,10 +21,24 @@ class cItemHandler
 {
 public:
 
-	enum eDurabilityLostAction
+	enum eDurabilityLostAction : int
```
[the remark this answers is not in the document]
Comment (bearbin) [reply src/Items/ItemHandler.h]: That would probably make sense, I just included it here as that's where it was before.

**Comment by bearbin** [comment]
> Overall I'm not sure it makes sense to add so many different categories of item use, especially when most only make sense for one item. Shouldn't all knowledge about particular items be in their own handler?

I agree the system as it is is rather convoluted and there are too many categories of item use, but I couldn't really think of a much better solution. If you have any ideas I'd be glad to hear them.

File: src/Items/ItemElytra.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
+
+// ItemElytra.h
+
+// Declares the various flint and steel ItemHandlers
```
[the remark this answers is not in the document]
Comment (bearbin) [reply src/Items/ItemElytra.h]: In fact, this comment is completle useless, I'm going to get rid of it on the other item handlers as well (most have no comment at all).

**Comment by peterbell10** [comment]
Some of the actions only happen when an item is used normally:  `dlaCarrotOnStickBoost`, `dlaElytraFly`, `dlaShootArrow`, `dlaTilling` and all the fishing ones.  I would think these could be handled directly within the item handlers' `OnItemUse` functions.

`dlaShearBlock` is only used for shears where it's the exact same damage as the other break block actions.  I would just not add it at all.

`dlaLightFire` and `dlaShearSheep` are both handling using an item on an entity.  Maybe give them a more generic name like `dlaUseEntity`, in which case they can share the same action.

This leaves `dlaShieldBlock` which doesn't really fit into `GetDurabilityLossByAction` because as the wiki says:
>When the shield blocks an attack, it takes damage equal to the strength of the attack (rounded down) plus 1.

**Comment by bearbin** [comment]
I believe that I have now addressed your comments. I removed all the damage categories except where the damage taken for an action could depend on the item used.

File: src/Entities/Player.h [hunk 3]
```diff
@@ -413,7 +414,16 @@ class cPlayer :
 	If the player is not riding a horse or if the horse is untamed, does nothing. */
 	void OpenHorseInventory();
 
-	void UseEquippedItem(int a_Amount = 1);
+	/** Damage the player's equipped item by a_Damage, possibly less if the
+	equipped item is enchanted. */
+	void UseEquippedItem(short a_Damage);
+
+	/** Shorthand for UseEquippedItem(1) */
+	void UseEquippedItem(void);
```
Comment (peterbell10) [inline src/Entities/Player.h]: What was wrong with using a default value?

Comment (bearbin) [reply src/Entities/Player.h]: That's a much better idea, I didn't think about the possibility.

**Pull request merged**